ui/src/app/views/account-creation/component.html

```diff
@@ -0,0 +1,55 @@
+<div style="padding: 1rem;">
+  <div *ngIf="!accountCreated" class="form-area">
+    <div class="form-section">
+      <label>Given Name:</label>
+      <input type="text" class="input" autofocus [(ngModel)]="givenName">
```

Comment:
For the model, can we instantiate a user/account request and use the fields in the user account request for all of the model values? Is there a reason not to do that?

Reply:
Thanks for asking this question. I tend to think of the model for a view as separate from its usage. Suppose this were a create/edit profile page. It would need to store fields like first name, last name, etc., but only when the user has finished filling out the form and takes action does it turn into a request to create something, update something, etc., where each payload which may be different. However, I do hate the duplication here. If we were using maps instead of objects, I could eliminate it (build the request by selecting certain fields from the map), but I'm not sure how to do that with typed objects.

Reply:
Is there a reason on the API side why a AccountCreationRequest is different from a User with not every field filled out? Could we make it so that in all cases it is just a user. (And when the account is being created, its just a user that hasn't gone into the database yet?) I don't know if there is a strong reason not to do that? (It mirrors what we do when creating workspaces as well)

Reply:
I like this idea. I will investigate this in a follow-up.